okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java

```diff
@@ -100,6 +100,9 @@ class OkHttpClientStream extends AbstractClientStream {
             transport,
             initialWindowSize,
             method.getFullMethodName());
+    if (callOptions.getOnReadyThreshold() != null) {
```

Comment:
We can do this in AbstractClientStream's constructor instead of each individual stream implementation. It is already receiving the `callOptions`. Moving the code there also allows you to delete the code that does this in AbstractClientStreamTest (because AbstractClientStream will be doing it already).

Reply:
I tried doing this in AbstractClientStream's constructor before and it we'd run into problems because setting the threshold depends on the _derived class's_ transport state, which is initialized in the derived constructor.

We could change this code to track the intended threshold then wait until start() is called to actually set it.

Reply:
Oh, I see. I remembered server-side where TransportState was created before the stream. On client-side it is created in reverse...

Can we pass `callOptions` to TransportState's constructor instead? Main reason to avoid that is if it means fixing a lot of tests, but if it isn't too painful, it is advantageous.